recurring_contract/models/recurring_contract.py

```diff
@@ -245,7 +269,7 @@ def write(self, vals):
         return res
 
     def unlink(self):
-        if self.filtered('start_date'):
+        if self.filtered('start_date') and not self.env.context.get("force_delete"):
```

Comment:
I know you didn't change that but is the start_date the condition for a validated contract ? Isn't the state the field we should test on ?

Reply:
The state can be changed at any time, the start_date is set once when we first validate the contract.